{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import sys\n",
    "from ftplib import FTP #import the ftp library\n",
    "import re \n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#cwd = os.getcwd()\n",
    "#print cwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "## different classes\n",
    "#t='Dinophyceae'\n",
    "#t='Bacillariophyta'\n",
    "t='Haptophyta'\n",
    "#t='Raphidophyceae'\n",
    "need='mmetsp_taxonomy.txt'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### MAFFT and HMMbuild in shell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "###run mafft and hmmbuild first to make script faster\n",
    "#os.system('./mafft_hmmbuil.sh')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Species list for HMMFUNCTION"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "will not retrive files, but will create a list with names that we can pass to bash and other functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "set(['Exanthemachrysis', 'Gephyrocapsa', 'Chrysochromulina', 'Isochrysis', 'Pleurochrysis', 'Pavlova', 'Phaeocystis', 'Coccolithus', 'Imantonia', 'Unidentified eukaryote', 'Prymnesium', 'Chrysoculter', 'Scyphosphaera', 'Emiliania', 'Calcidiscus'])\n"
     ]
    }
   ],
   "source": [
    "mt=open('mmetsp_taxonomy.txt','r')\n",
    "g=[] #make an empty list to store genus names\n",
    "for line in mt:\n",
    "    if re.search(t,line): #if taxa name in line\n",
    "        g= g+line.split('\\t')[7:8]#pull out the 8th field should be genus, keeping as list\n",
    "\n",
    "g=set(g) #keep only unique genus names\n",
    "print g\n",
    "#close the taxonomy file\n",
    "mt.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Gephyrocapsa-oceanica-RCC1303.pep.fa.gz', 'Chrysochromulina-polylepis-CCMP1757.pep.fa.gz', 'Isochrysis-galbana-CCMP1323.pep.fa.gz', 'Isochrysis-sp-CCMP1244.pep.fa.gz', 'Isochrysis-sp-CCMP1324.pep.fa.gz', 'Pleurochrysis-carterae-CCMP645.pep.fa.gz', 'Pavlova-sp-CCMP459.pep.fa.gz', 'Prymnesium-parvum-Texoma1.pep.fa.gz', 'Emiliania-huxleyi-374.pep.fa.gz', 'Emiliania-huxleyi-379.pep.fa.gz', 'Emiliania-huxleyi-CCMP370.pep.fa.gz', 'Emiliania-huxleyi-PLYM219.pep.fa.gz']\n"
     ]
    }
   ],
   "source": [
    "ftp= FTP('ftp.imicrobe.us') #set home ftp server\n",
    "ftp.login() #log in\n",
    "ftp.cwd('camera/combined_assemblies') #ch\n",
    "\n",
    "files=ftp.nlst() #make a list of all files and directories in wd\n",
    "delimiter=' '\n",
    "all=delimiter.join(files)\n",
    "\n",
    "names=[]\n",
    "\n",
    "for genus in g:\n",
    "    string= genus+\"\\S*.pep.fa.gz\"\n",
    "    taxafiles=re.findall(string, all)\n",
    "    #print \"{} files matching genus=\".format(len(taxafiles))+genus\n",
    "    #print taxafiles\n",
    "    if len(taxafiles) > 0:\n",
    "        for filex in taxafiles:\n",
    "            command = \"RETR \"+filex\n",
    "            outfile = filex\n",
    "            #ftp.retrbinary(command, open(outfile, 'wb').write)\n",
    "            names.append(outfile)\n",
    "            \n",
    "ftp.quit()\n",
    "\n",
    "print names\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Gephyrocapsa-oceanica-RCC1303.pep.fa.gz', 'Chrysochromulina-polylepis-CCMP1757.pep.fa.gz']\n"
     ]
    }
   ],
   "source": [
    "names=names[:2]\n",
    "print names"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Retrieve PEP.fa in shell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "###run retriver in shell\n",
    "os.system('python ./MMETSP_sample_import.py {} {}'.format(t,need))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Fix error on Dinos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30\n",
      "27\n"
     ]
    }
   ],
   "source": [
    "##the last species is not available for some reason so I removed it\n",
    "print len(names)\n",
    "\n",
    "if t=='Dinophyceae':\n",
    "    names.remove('Durinskia-baltica-CSIRO_CS-38.pep.fa.gz')\n",
    "    names.remove('Oxyrrhis-marina-CCMP1795.pep.fa.gz')\n",
    "    names.remove('Alexandrium-fundyense-CCMP1719.pep.fa.gz')\n",
    "    \n",
    "print len(names)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### GET counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "names=[i.split('.')[0] for i in names] #removes .pep.fa.gz from the names\n",
    "\n",
    "ftp= FTP('ftp.imicrobe.us') #set ftp server\n",
    "ftp.login() #log in\n",
    "ftp.cwd('camera/combined_assemblies') #change directory\n",
    "\n",
    "t='/Users/maria_hernandez/Documents/Big_Data3050/CMM_MoreSP/'  #location for files\n",
    "for ID in names:\n",
    "    #change to taxa directory/readcounts\n",
    "    ripdir= ID+\"/readcounts\"\n",
    "    ftp.cwd(ripdir) #change directory\n",
    "    savefile= t+ID+\"_cds_counts.txt\" #saves files with unique names\n",
    "    ftp.retrbinary('RETR cds.dat', open(savefile, 'wb').write)\n",
    "    ftp.cwd(\"~/camera/combined_assemblies\") #change directory to restart loop in right place\n",
    "    \n",
    "ftp.quit()   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "names='Chrysochromulina-polylepis-CCMP1757.pep.fa.gz'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### RUN HMM in shell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "for i in names:\n",
    "    os.system('./MAGIC_HMM.sh {}'.format(i))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for i in names:\n",
    "    os.system('./MAGIC_one.sh {}'.format(i))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### HMM READ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def readHMM(Organism,gene_name):\n",
    "    \"\"\" Takes in organism and gene_name from HMM results and makes a table.\n",
    "    HMM results from --tblout that have the following name organism_genename_HMM.csv\n",
    "    Note: pep.fa files differ in structure and it can affect how the HMM output is written. If you can't read the file in \n",
    "    modify the fuction\"\"\"\n",
    "\n",
    "    hold=pd.read_csv('{}_{}_HMM.csv'.format(Organism,gene_name),delim_whitespace=True,engine='python',skiprows=2,skipfooter=10)\n",
    "    #empty files have 12 rows so the following if statement will only work on files that are not empty\n",
    "    \n",
    "    if hold.shape[0]!=0:\n",
    "        readX=pd.read_csv('{}_{}_HMM.csv'.format(Organism,gene_name),delim_whitespace=True,engine='python',skiprows=3,\n",
    "                          header=None ,skipfooter=10)\n",
    "        \n",
    "        samplenames=[]\n",
    "        for i in np.arange(1,readX.shape[1]+1):\n",
    "            samplenames.append(str(i))\n",
    "\n",
    "        readX.columns=samplenames\n",
    "        \n",
    "        new=pd.DataFrame()\n",
    "        \n",
    "        new['CAMPEPid']=readX['1']\n",
    "        new['contig']=readX['19']\n",
    "        new['Evalue']=readX['5']\n",
    "        new['Annotation']='{}'.format(gene_name)\n",
    "\n",
    "        new.contig=new.contig.str.split(\"|\").str[1]\n",
    "        new.contig=new.contig.str.split(\"_\").str[0]\n",
    "        return new\n",
    "        #return readX"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CAMPEPid</th>\n",
       "      <th>contig</th>\n",
       "      <th>Evalue</th>\n",
       "      <th>Annotation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAMPEP_0193794506</td>\n",
       "      <td>169964</td>\n",
       "      <td>1.600000e-125</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CAMPEP_0193783668</td>\n",
       "      <td>140936</td>\n",
       "      <td>1.900000e-44</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CAMPEP_0193795430</td>\n",
       "      <td>170428</td>\n",
       "      <td>1.400000e-43</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAMPEP_0193788850</td>\n",
       "      <td>155911</td>\n",
       "      <td>1.300000e-25</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CAMPEP_0193725542</td>\n",
       "      <td>15531</td>\n",
       "      <td>4.000000e-24</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            CAMPEPid  contig         Evalue  Annotation\n",
       "0  CAMPEP_0193794506  169964  1.600000e-125  Bestrophin\n",
       "1  CAMPEP_0193783668  140936   1.900000e-44  Bestrophin\n",
       "2  CAMPEP_0193795430  170428   1.400000e-43  Bestrophin\n",
       "3  CAMPEP_0193788850  155911   1.300000e-25  Bestrophin\n",
       "4  CAMPEP_0193725542   15531   4.000000e-24  Bestrophin"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "readHMM('Chrysochromulina-polylepis-CCMP1757.pep.fa.gz','Bestrophin').head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def makeHMM(Organism,gene_type):\n",
    "    \"\"\"Combines the plots from each gene_type into a table\"\"\"\n",
    "    if gene_type=='CF':\n",
    "        a=readHMM(Organism,'PK')\n",
    "        b=readHMM(Organism,'PEPC')\n",
    "        c=readHMM(Organism,'PEPCK')\n",
    "        d=readHMM(Organism,'PK')\n",
    "        e=readHMM(Organism,'MDH')\n",
    "        f=readHMM(Organism,'OMT')\n",
    "        g=readHMM(Organism,'PYC')\n",
    "        h=readHMM(Organism,'PPDK')\n",
    "        i=readHMM(Organism,'ME')\n",
    "        \n",
    "        frames = [a,b,c,d,e,f,g,i,h]\n",
    "        result = pd.concat(frames)\n",
    "        return result\n",
    "    if gene_type=='PR':\n",
    "        a=readHMM(Organism,'SHMT')\n",
    "        b=readHMM(Organism,'GOX')\n",
    "        c=readHMM(Organism,'GDCT')\n",
    "        d=readHMM(Organism,'PGP')\n",
    "        e=readHMM(Organism,'ICL')\n",
    "        f=readHMM(Organism,'GCL')\n",
    "        g=readHMM(Organism,'HR')\n",
    "        h=readHMM(Organism,'SPT')\n",
    "        i=readHMM(Organism,'TSR')\n",
    "        j=readHMM(Organism,'MS')\n",
    "        k=readHMM(Organism,'GlcDH')\n",
    "        l=readHMM(Organism,'ALAT_GGAT')\n",
    "        m=readHMM(Organism,'GK')\n",
    "        \n",
    "        frames = [a,b,c,d,e,f,g,h,i,j,k,l,m]\n",
    "        result = pd.concat(frames)\n",
    "        return result\n",
    "    if gene_type=='BP':\n",
    "        a=readHMM(Organism,'CA_alpha')\n",
    "        b=readHMM(Organism,'CA_delta')\n",
    "        c=readHMM(Organism,'CA_beta')\n",
    "        #d=readHMM(Organism,'Ca_zeta')\n",
    "        e=readHMM(Organism,'Bestrophin')\n",
    "        f=readHMM(Organism,'SLC4')\n",
    "\n",
    "        frames = [a,b,c,e,f]\n",
    "        result = pd.concat(frames)\n",
    "        return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CAMPEPid</th>\n",
       "      <th>contig</th>\n",
       "      <th>Evalue</th>\n",
       "      <th>Annotation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAMPEP_0193716998</td>\n",
       "      <td>10130_1</td>\n",
       "      <td>8.300000e-164</td>\n",
       "      <td>CA_delta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CAMPEP_0193720114</td>\n",
       "      <td>12139_1</td>\n",
       "      <td>2.500000e-144</td>\n",
       "      <td>CA_delta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CAMPEP_0193740950</td>\n",
       "      <td>25073_1</td>\n",
       "      <td>2.000000e-112</td>\n",
       "      <td>CA_delta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAMPEP_0193744210</td>\n",
       "      <td>27355_1</td>\n",
       "      <td>1.700000e-97</td>\n",
       "      <td>CA_delta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAMPEP_0193731632</td>\n",
       "      <td>19523_1</td>\n",
       "      <td>7.400000e-17</td>\n",
       "      <td>CA_beta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CAMPEP_0193739048</td>\n",
       "      <td>23878_1</td>\n",
       "      <td>4.500000e-07</td>\n",
       "      <td>CA_beta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAMPEP_0193794506</td>\n",
       "      <td>169964_1</td>\n",
       "      <td>1.600000e-125</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CAMPEP_0193783668</td>\n",
       "      <td>140936_1</td>\n",
       "      <td>1.900000e-44</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CAMPEP_0193795430</td>\n",
       "      <td>170428_1</td>\n",
       "      <td>1.400000e-43</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAMPEP_0193788850</td>\n",
       "      <td>155911_1</td>\n",
       "      <td>1.300000e-25</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CAMPEP_0193725542</td>\n",
       "      <td>15531_1</td>\n",
       "      <td>4.000000e-24</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CAMPEP_0193705254</td>\n",
       "      <td>1979_1</td>\n",
       "      <td>2.600000e-23</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CAMPEP_0193801006</td>\n",
       "      <td>173214_1</td>\n",
       "      <td>5.100000e-23</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CAMPEP_0193712428</td>\n",
       "      <td>6963_1</td>\n",
       "      <td>7.900000e-23</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CAMPEP_0193742798</td>\n",
       "      <td>26324_1</td>\n",
       "      <td>1.400000e-21</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CAMPEP_0193727680</td>\n",
       "      <td>16768_1</td>\n",
       "      <td>7.400000e-21</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CAMPEP_0193741146</td>\n",
       "      <td>25209_1</td>\n",
       "      <td>4.600000e-20</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CAMPEP_0193737912</td>\n",
       "      <td>23215_1</td>\n",
       "      <td>3.300000e-18</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CAMPEP_0193731534</td>\n",
       "      <td>19445_1</td>\n",
       "      <td>1.400000e-12</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CAMPEP_0193704942</td>\n",
       "      <td>1794_1</td>\n",
       "      <td>3.800000e-12</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CAMPEP_0193706960</td>\n",
       "      <td>3047_1</td>\n",
       "      <td>5.300000e-12</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CAMPEP_0193714820</td>\n",
       "      <td>8398_1</td>\n",
       "      <td>5.500000e-12</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CAMPEP_0193764498</td>\n",
       "      <td>56150_1</td>\n",
       "      <td>1.500000e-11</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>CAMPEP_0193735246</td>\n",
       "      <td>21696_1</td>\n",
       "      <td>2.000000e-09</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>CAMPEP_0193761482</td>\n",
       "      <td>51549_1</td>\n",
       "      <td>2.400000e-09</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>CAMPEP_0193790988</td>\n",
       "      <td>168202_1</td>\n",
       "      <td>3.800000e-09</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>CAMPEP_0193785494</td>\n",
       "      <td>145252_1</td>\n",
       "      <td>1.300000e-08</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>CAMPEP_0193779272</td>\n",
       "      <td>125944_1</td>\n",
       "      <td>1.300000e-08</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>CAMPEP_0193779030</td>\n",
       "      <td>125485_1</td>\n",
       "      <td>1.500000e-08</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CAMPEP_0193709764</td>\n",
       "      <td>5146_1</td>\n",
       "      <td>1.600000e-08</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>CAMPEP_0193708378</td>\n",
       "      <td>4061_1</td>\n",
       "      <td>4.700000e-06</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>CAMPEP_0193783502</td>\n",
       "      <td>140102_1</td>\n",
       "      <td>5.000000e-06</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>CAMPEP_0193742954</td>\n",
       "      <td>26440_1</td>\n",
       "      <td>5.900000e-06</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>CAMPEP_0193721942</td>\n",
       "      <td>13485_1</td>\n",
       "      <td>8.700000e-06</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>CAMPEP_0193746848</td>\n",
       "      <td>31034_1</td>\n",
       "      <td>2.400000e-05</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>CAMPEP_0193796456</td>\n",
       "      <td>170945_1</td>\n",
       "      <td>3.800000e-05</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>CAMPEP_0193724924</td>\n",
       "      <td>15193_1</td>\n",
       "      <td>6.700000e-05</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>CAMPEP_0193792018</td>\n",
       "      <td>168721_1</td>\n",
       "      <td>7.800000e-05</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>CAMPEP_0193708728</td>\n",
       "      <td>4339_1</td>\n",
       "      <td>8.100000e-05</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>CAMPEP_0193739010</td>\n",
       "      <td>23853_1</td>\n",
       "      <td>3.300000e-04</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>CAMPEP_0193782824</td>\n",
       "      <td>136333_1</td>\n",
       "      <td>5.700000e-04</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>CAMPEP_0193800510</td>\n",
       "      <td>172967_1</td>\n",
       "      <td>7.400000e-04</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>CAMPEP_0193766222</td>\n",
       "      <td>61043_1</td>\n",
       "      <td>8.100000e-04</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>CAMPEP_0193732752</td>\n",
       "      <td>20295_1</td>\n",
       "      <td>1.300000e-03</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>CAMPEP_0193760434</td>\n",
       "      <td>50060_1</td>\n",
       "      <td>6.400000e-03</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>CAMPEP_0193703674</td>\n",
       "      <td>1074_1</td>\n",
       "      <td>6.800000e-03</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>CAMPEP_0193760374</td>\n",
       "      <td>49940_1</td>\n",
       "      <td>1.000000e-02</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>CAMPEP_0193759686</td>\n",
       "      <td>48915_1</td>\n",
       "      <td>2.100000e-02</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>CAMPEP_0193752988</td>\n",
       "      <td>39679_1</td>\n",
       "      <td>4.200000e-02</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>CAMPEP_0193764752</td>\n",
       "      <td>56822_1</td>\n",
       "      <td>6.800000e-02</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>CAMPEP_0193741608</td>\n",
       "      <td>25516_1</td>\n",
       "      <td>9.000000e-02</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>CAMPEP_0193800210</td>\n",
       "      <td>172820_1</td>\n",
       "      <td>2.000000e-01</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>CAMPEP_0193738984</td>\n",
       "      <td>23837_1</td>\n",
       "      <td>1.300000e+00</td>\n",
       "      <td>Bestrophin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAMPEP_0193734326</td>\n",
       "      <td>21185_1</td>\n",
       "      <td>1.100000e-131</td>\n",
       "      <td>SLC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CAMPEP_0193793958</td>\n",
       "      <td>169690_1</td>\n",
       "      <td>1.200000e-86</td>\n",
       "      <td>SLC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CAMPEP_0193739320</td>\n",
       "      <td>24061_1</td>\n",
       "      <td>6.400000e-79</td>\n",
       "      <td>SLC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAMPEP_0193749936</td>\n",
       "      <td>35400_1</td>\n",
       "      <td>1.900000e-17</td>\n",
       "      <td>SLC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CAMPEP_0193785558</td>\n",
       "      <td>145391_1</td>\n",
       "      <td>5.900000e-14</td>\n",
       "      <td>SLC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CAMPEP_0193709316</td>\n",
       "      <td>4808_1</td>\n",
       "      <td>3.300000e+00</td>\n",
       "      <td>SLC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CAMPEP_0193780838</td>\n",
       "      <td>129005_1</td>\n",
       "      <td>3.500000e+00</td>\n",
       "      <td>SLC4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>80 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             CAMPEPid    contig         Evalue  Annotation\n",
       "0   CAMPEP_0193716998   10130_1  8.300000e-164    CA_delta\n",
       "1   CAMPEP_0193720114   12139_1  2.500000e-144    CA_delta\n",
       "2   CAMPEP_0193740950   25073_1  2.000000e-112    CA_delta\n",
       "3   CAMPEP_0193744210   27355_1   1.700000e-97    CA_delta\n",
       "0   CAMPEP_0193731632   19523_1   7.400000e-17     CA_beta\n",
       "1   CAMPEP_0193739048   23878_1   4.500000e-07     CA_beta\n",
       "0   CAMPEP_0193794506  169964_1  1.600000e-125  Bestrophin\n",
       "1   CAMPEP_0193783668  140936_1   1.900000e-44  Bestrophin\n",
       "2   CAMPEP_0193795430  170428_1   1.400000e-43  Bestrophin\n",
       "3   CAMPEP_0193788850  155911_1   1.300000e-25  Bestrophin\n",
       "4   CAMPEP_0193725542   15531_1   4.000000e-24  Bestrophin\n",
       "5   CAMPEP_0193705254    1979_1   2.600000e-23  Bestrophin\n",
       "6   CAMPEP_0193801006  173214_1   5.100000e-23  Bestrophin\n",
       "7   CAMPEP_0193712428    6963_1   7.900000e-23  Bestrophin\n",
       "8   CAMPEP_0193742798   26324_1   1.400000e-21  Bestrophin\n",
       "9   CAMPEP_0193727680   16768_1   7.400000e-21  Bestrophin\n",
       "10  CAMPEP_0193741146   25209_1   4.600000e-20  Bestrophin\n",
       "11  CAMPEP_0193737912   23215_1   3.300000e-18  Bestrophin\n",
       "12  CAMPEP_0193731534   19445_1   1.400000e-12  Bestrophin\n",
       "13  CAMPEP_0193704942    1794_1   3.800000e-12  Bestrophin\n",
       "14  CAMPEP_0193706960    3047_1   5.300000e-12  Bestrophin\n",
       "15  CAMPEP_0193714820    8398_1   5.500000e-12  Bestrophin\n",
       "16  CAMPEP_0193764498   56150_1   1.500000e-11  Bestrophin\n",
       "17  CAMPEP_0193735246   21696_1   2.000000e-09  Bestrophin\n",
       "18  CAMPEP_0193761482   51549_1   2.400000e-09  Bestrophin\n",
       "19  CAMPEP_0193790988  168202_1   3.800000e-09  Bestrophin\n",
       "20  CAMPEP_0193785494  145252_1   1.300000e-08  Bestrophin\n",
       "21  CAMPEP_0193779272  125944_1   1.300000e-08  Bestrophin\n",
       "22  CAMPEP_0193779030  125485_1   1.500000e-08  Bestrophin\n",
       "23  CAMPEP_0193709764    5146_1   1.600000e-08  Bestrophin\n",
       "..                ...       ...            ...         ...\n",
       "44  CAMPEP_0193708378    4061_1   4.700000e-06  Bestrophin\n",
       "45  CAMPEP_0193783502  140102_1   5.000000e-06  Bestrophin\n",
       "46  CAMPEP_0193742954   26440_1   5.900000e-06  Bestrophin\n",
       "47  CAMPEP_0193721942   13485_1   8.700000e-06  Bestrophin\n",
       "48  CAMPEP_0193746848   31034_1   2.400000e-05  Bestrophin\n",
       "49  CAMPEP_0193796456  170945_1   3.800000e-05  Bestrophin\n",
       "50  CAMPEP_0193724924   15193_1   6.700000e-05  Bestrophin\n",
       "51  CAMPEP_0193792018  168721_1   7.800000e-05  Bestrophin\n",
       "52  CAMPEP_0193708728    4339_1   8.100000e-05  Bestrophin\n",
       "53  CAMPEP_0193739010   23853_1   3.300000e-04  Bestrophin\n",
       "54  CAMPEP_0193782824  136333_1   5.700000e-04  Bestrophin\n",
       "55  CAMPEP_0193800510  172967_1   7.400000e-04  Bestrophin\n",
       "56  CAMPEP_0193766222   61043_1   8.100000e-04  Bestrophin\n",
       "57  CAMPEP_0193732752   20295_1   1.300000e-03  Bestrophin\n",
       "58  CAMPEP_0193760434   50060_1   6.400000e-03  Bestrophin\n",
       "59  CAMPEP_0193703674    1074_1   6.800000e-03  Bestrophin\n",
       "60  CAMPEP_0193760374   49940_1   1.000000e-02  Bestrophin\n",
       "61  CAMPEP_0193759686   48915_1   2.100000e-02  Bestrophin\n",
       "62  CAMPEP_0193752988   39679_1   4.200000e-02  Bestrophin\n",
       "63  CAMPEP_0193764752   56822_1   6.800000e-02  Bestrophin\n",
       "64  CAMPEP_0193741608   25516_1   9.000000e-02  Bestrophin\n",
       "65  CAMPEP_0193800210  172820_1   2.000000e-01  Bestrophin\n",
       "66  CAMPEP_0193738984   23837_1   1.300000e+00  Bestrophin\n",
       "0   CAMPEP_0193734326   21185_1  1.100000e-131        SLC4\n",
       "1   CAMPEP_0193793958  169690_1   1.200000e-86        SLC4\n",
       "2   CAMPEP_0193739320   24061_1   6.400000e-79        SLC4\n",
       "3   CAMPEP_0193749936   35400_1   1.900000e-17        SLC4\n",
       "4   CAMPEP_0193785558  145391_1   5.900000e-14        SLC4\n",
       "5   CAMPEP_0193709316    4808_1   3.300000e+00        SLC4\n",
       "6   CAMPEP_0193780838  129005_1   3.500000e+00        SLC4\n",
       "\n",
       "[80 rows x 4 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "makeHMM('Chrysochromulina-polylepis-CCMP1757.pep.fa.gz','BP')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def HMMcmp(Organism,gene_type):\n",
    "    result=makeHMM(Organism,gene_type)\n",
    "    contig_EV=dict(zip(result.contig,result.Evalue))\n",
    "    contig_Annot=dict(zip(result.contig,result.Annotation))\n",
    "    \n",
    "    \n",
    "    Organism = re.sub('.pep.fa.gz', '', Organism)\n",
    "    \n",
    "    expression=pd.read_csv('{}_cds_counts.txt'.format(Organism),delimiter='\\t',index_col=0)\n",
    " \n",
    "    expression['log2CPM']=np.log2(expression.sum(axis=1)/1000000)\n",
    "    expression.index= expression.index.str.split(\"|\").str[1]\n",
    "    contig_CMP=dict(zip(expression.index,expression.log2CPM))\n",
    "    \n",
    "    Contig=[]\n",
    "    Evalue=[]\n",
    "    Annotation=[]\n",
    "    log2CPM=[]\n",
    "    \n",
    "    for i in expression.index:\n",
    "        a=contig_CMP.get(i)\n",
    "        if a>-16 or a==\"-inf\" or a==\"inf\":\n",
    "            Contig.append(i)\n",
    "            Evalue.append(contig_EV.get(i))\n",
    "            Annotation.append(contig_Annot.get(i))\n",
    "            log2CPM.append(a)\n",
    "              \n",
    "\n",
    "    out=pd.DataFrame()\n",
    "    out['Contig']=Contig\n",
    "    out['Evalue']=Evalue\n",
    "    out['Annotation']=Annotation\n",
    "    out['log2CPM']=log2CPM\n",
    "        \n",
    "    return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Contig</th>\n",
       "      <th>Evalue</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>log2CPM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-6.713308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-6.339929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.924541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-14.844106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.441721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>8_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.127438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-11.431723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-7.236558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-15.609640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-15.231129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.400187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>16_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-8.201098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>15_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-6.864807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>18_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-12.488625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>17_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.552190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>20_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-14.761644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>23_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-12.823044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>14_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-7.118589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>22_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-7.907814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>13_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.756643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>24_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.447753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>26_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-11.150209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>25_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-13.439715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>28_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-15.609640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>27_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.931569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>30_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-11.055052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>31_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-12.061204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>33_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.766662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>35_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-14.761644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>36_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.797142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49525</th>\n",
       "      <td>173613_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.884445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49526</th>\n",
       "      <td>173610_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-8.018679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49527</th>\n",
       "      <td>173614_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.277828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49528</th>\n",
       "      <td>173615_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.524301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49529</th>\n",
       "      <td>173616_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.744216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49530</th>\n",
       "      <td>173617_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-8.363612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49531</th>\n",
       "      <td>173594_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.517941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49532</th>\n",
       "      <td>173619_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-11.000831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49533</th>\n",
       "      <td>173618_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.690777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49534</th>\n",
       "      <td>173620_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.180025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49535</th>\n",
       "      <td>173621_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-7.328406</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49536</th>\n",
       "      <td>173623_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.746693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49537</th>\n",
       "      <td>173622_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.873577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49538</th>\n",
       "      <td>173624_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.124214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49539</th>\n",
       "      <td>173625_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-5.489727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49540</th>\n",
       "      <td>173628_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.125825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49541</th>\n",
       "      <td>173626_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-8.247258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49542</th>\n",
       "      <td>173629_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.190102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49543</th>\n",
       "      <td>173631_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-7.165661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49544</th>\n",
       "      <td>173632_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-11.231129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49545</th>\n",
       "      <td>173633_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.676540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49546</th>\n",
       "      <td>173634_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-6.513189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49547</th>\n",
       "      <td>173627_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.186735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49548</th>\n",
       "      <td>173635_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-10.196859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49549</th>\n",
       "      <td>173636_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.434715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49550</th>\n",
       "      <td>173637_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-8.188417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49551</th>\n",
       "      <td>173638_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-5.760235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49552</th>\n",
       "      <td>173639_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-2.661282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49553</th>\n",
       "      <td>173554_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-7.592275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49554</th>\n",
       "      <td>173630_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-8.716036</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49555 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Contig  Evalue Annotation    log2CPM\n",
       "0           3_1     NaN       None  -6.713308\n",
       "1           1_1     NaN       None  -6.339929\n",
       "2           4_1     NaN       None  -9.924541\n",
       "3           6_1     NaN       None -14.844106\n",
       "4           5_1     NaN       None  -9.441721\n",
       "5           8_1     NaN       None -10.127438\n",
       "6           7_1     NaN       None -11.431723\n",
       "7           2_1     NaN       None  -7.236558\n",
       "8           9_1     NaN       None -15.609640\n",
       "9          10_1     NaN       None -15.231129\n",
       "10         11_1     NaN       None -10.400187\n",
       "11         16_1     NaN       None  -8.201098\n",
       "12         15_1     NaN       None  -6.864807\n",
       "13         18_1     NaN       None -12.488625\n",
       "14         17_1     NaN       None -10.552190\n",
       "15         20_1     NaN       None -14.761644\n",
       "16         23_1     NaN       None -12.823044\n",
       "17         14_1     NaN       None  -7.118589\n",
       "18         22_1     NaN       None  -7.907814\n",
       "19         13_1     NaN       None -10.756643\n",
       "20         24_1     NaN       None  -9.447753\n",
       "21         26_1     NaN       None -11.150209\n",
       "22         25_1     NaN       None -13.439715\n",
       "23         28_1     NaN       None -15.609640\n",
       "24         27_1     NaN       None -10.931569\n",
       "25         30_1     NaN       None -11.055052\n",
       "26         31_1     NaN       None -12.061204\n",
       "27         33_1     NaN       None -10.766662\n",
       "28         35_1     NaN       None -14.761644\n",
       "29         36_1     NaN       None -10.797142\n",
       "...         ...     ...        ...        ...\n",
       "49525  173613_1     NaN       None  -9.884445\n",
       "49526  173610_1     NaN       None  -8.018679\n",
       "49527  173614_1     NaN       None  -9.277828\n",
       "49528  173615_1     NaN       None  -9.524301\n",
       "49529  173616_1     NaN       None  -9.744216\n",
       "49530  173617_1     NaN       None  -8.363612\n",
       "49531  173594_1     NaN       None -10.517941\n",
       "49532  173619_1     NaN       None -11.000831\n",
       "49533  173618_1     NaN       None  -9.690777\n",
       "49534  173620_1     NaN       None -10.180025\n",
       "49535  173621_1     NaN       None  -7.328406\n",
       "49536  173623_1     NaN       None -10.746693\n",
       "49537  173622_1     NaN       None  -9.873577\n",
       "49538  173624_1     NaN       None -10.124214\n",
       "49539  173625_1     NaN       None  -5.489727\n",
       "49540  173628_1     NaN       None  -9.125825\n",
       "49541  173626_1     NaN       None  -8.247258\n",
       "49542  173629_1     NaN       None -10.190102\n",
       "49543  173631_1     NaN       None  -7.165661\n",
       "49544  173632_1     NaN       None -11.231129\n",
       "49545  173633_1     NaN       None  -9.676540\n",
       "49546  173634_1     NaN       None  -6.513189\n",
       "49547  173627_1     NaN       None -10.186735\n",
       "49548  173635_1     NaN       None -10.196859\n",
       "49549  173636_1     NaN       None  -9.434715\n",
       "49550  173637_1     NaN       None  -8.188417\n",
       "49551  173638_1     NaN       None  -5.760235\n",
       "49552  173639_1     NaN       None  -2.661282\n",
       "49553  173554_1     NaN       None  -7.592275\n",
       "49554  173630_1     NaN       None  -8.716036\n",
       "\n",
       "[49555 rows x 4 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HMMcmp('Chrysochromulina-polylepis-CCMP1757.pep.fa.gz','BP')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def HMMclean(Organism):\n",
    "    '''Will create an HMM table and filter out values that do not meet an e-value'''\n",
    "    a=HMMcmp(Organism,'BP')\n",
    "    b=HMMcmp(Organism,'CF')\n",
    "    c=HMMcmp(Organism,'PR')\n",
    "    \n",
    "    frames=[a,b,c]\n",
    "    HMM= pd.concat(frames)\n",
    "    \n",
    "    #print HMM\n",
    "    \n",
    "    HMM=HMM[HMM['Evalue'] < .00001]\n",
    "    \n",
    "    #return HMM.drop_duplicates()\n",
    "    \n",
    "    Genes= np.unique(HMM['Annotation'])\n",
    "    A=pd.DataFrame(0,index=[Organism],columns=Genes)\n",
    "    \n",
    "    for j in Genes:\n",
    "        counts=HMM.Annotation.value_counts()[j]\n",
    "        A[j]= counts\n",
    "           \n",
    "    return A\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ALAT_GGAT</th>\n",
       "      <th>Bestrophin</th>\n",
       "      <th>CA_beta</th>\n",
       "      <th>CA_delta</th>\n",
       "      <th>GCL</th>\n",
       "      <th>GDCT</th>\n",
       "      <th>GlcDH</th>\n",
       "      <th>HR</th>\n",
       "      <th>ICL</th>\n",
       "      <th>MDH</th>\n",
       "      <th>...</th>\n",
       "      <th>OMT</th>\n",
       "      <th>PEPC</th>\n",
       "      <th>PEPCK</th>\n",
       "      <th>PGP</th>\n",
       "      <th>PK</th>\n",
       "      <th>PYC</th>\n",
       "      <th>SHMT</th>\n",
       "      <th>SLC4</th>\n",
       "      <th>SPT</th>\n",
       "      <th>TSR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Chrysochromulina-polylepis-CCMP1757.pep.fa.gz</th>\n",
       "      <td>8</td>\n",
       "      <td>48</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>47</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               ALAT_GGAT  Bestrophin  CA_beta  \\\n",
       "Chrysochromulina-polylepis-CCMP1757.pep.fa.gz          8          48        2   \n",
       "\n",
       "                                               CA_delta  GCL  GDCT  GlcDH  HR  \\\n",
       "Chrysochromulina-polylepis-CCMP1757.pep.fa.gz         4    2     1      4   9   \n",
       "\n",
       "                                               ICL  MDH ...   OMT  PEPC  \\\n",
       "Chrysochromulina-polylepis-CCMP1757.pep.fa.gz    3    2 ...    47     1   \n",
       "\n",
       "                                               PEPCK  PGP  PK  PYC  SHMT  \\\n",
       "Chrysochromulina-polylepis-CCMP1757.pep.fa.gz      1   11   9   10     4   \n",
       "\n",
       "                                               SLC4  SPT  TSR  \n",
       "Chrysochromulina-polylepis-CCMP1757.pep.fa.gz     5    3    7  \n",
       "\n",
       "[1 rows x 21 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HMMclean('Chrysochromulina-polylepis-CCMP1757.pep.fa.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Gephyrocapsa-oceanica-RCC1303.pep.fa.gz', 'Chrysochromulina-polylepis-CCMP1757.pep.fa.gz']\n"
     ]
    }
   ],
   "source": [
    "print names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def gen_type(gene_type):\n",
    "    \"\"\" Allows you to extract specific gene_types\"\"\"\n",
    "    \n",
    "    if gene_type=='BP':\n",
    "            fixed=['SLC4','Bestrophin','CA_alpha','CA_beta','CA_delta','CA_zeta']\n",
    "    if gene_type=='PR':\n",
    "            fixed=['PGP','GOX','SPT','ALAT_GGAT','GDCT','SHMT','HR','GK','GlcDH','MS','ICL','GCL','TSR']\n",
    "    if gene_type=='CF':\n",
    "            fixed=['PK','PEPC','PEPCK','MDH','OMT','ME','PPDK','PYC']\n",
    "            \n",
    "    A=pd.DataFrame(0, index=names, columns=fixed)\n",
    "    \n",
    "    for i,j in enumerate(names):\n",
    "        for k in fixed:\n",
    "            B=HMMclean(j)\n",
    "            #print B\n",
    "            if k in list(B):\n",
    "                A[k][i]=B[k]\n",
    "                \n",
    "    A.index = A.index.str.split('.').str[0]\n",
    "    \n",
    "    C=A.transpose()\n",
    "    #C.to_csv('{}_{}_GeneCountHMM.csv'.format(t,gene_type))\n",
    "    \n",
    "    return A.transpose()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#gen_type('BP')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### All sections in one table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            Gephyrocapsa-oceanica-RCC1303  \\\n",
      "SLC4                                    6   \n",
      "Bestrophin                             37   \n",
      "CA_alpha                                2   \n",
      "CA_beta                                 0   \n",
      "CA_delta                                2   \n",
      "CA_zeta                                 0   \n",
      "PK                                      4   \n",
      "PEPC                                    1   \n",
      "PEPCK                                   1   \n",
      "MDH                                     4   \n",
      "OMT                                    69   \n",
      "ME                                      2   \n",
      "PPDK                                    0   \n",
      "PYC                                    16   \n",
      "PGP                                     9   \n",
      "GOX                                     1   \n",
      "SPT                                     2   \n",
      "ALAT_GGAT                              13   \n",
      "GDCT                                    4   \n",
      "SHMT                                    4   \n",
      "HR                                     10   \n",
      "GK                                      0   \n",
      "GlcDH                                   5   \n",
      "MS                                      0   \n",
      "ICL                                     4   \n",
      "GCL                                     4   \n",
      "TSR                                     7   \n",
      "\n",
      "            Chrysochromulina-polylepis-CCMP1757  Isochrysis-galbana-CCMP1323  \\\n",
      "SLC4                                          5                           14   \n",
      "Bestrophin                                   48                           34   \n",
      "CA_alpha                                      0                            2   \n",
      "CA_beta                                       2                            7   \n",
      "CA_delta                                      4                            2   \n",
      "CA_zeta                                       0                            0   \n",
      "PK                                            9                            6   \n",
      "PEPC                                          1                            1   \n",
      "PEPCK                                         1                            2   \n",
      "MDH                                           2                            4   \n",
      "OMT                                          47                           75   \n",
      "ME                                            3                            8   \n",
      "PPDK                                          0                            0   \n",
      "PYC                                          10                           17   \n",
      "PGP                                          11                           11   \n",
      "GOX                                           0                            2   \n",
      "SPT                                           3                            6   \n",
      "ALAT_GGAT                                     8                           17   \n",
      "GDCT                                          1                            3   \n",
      "SHMT                                          4                            9   \n",
      "HR                                            9                            8   \n",
      "GK                                            0                            0   \n",
      "GlcDH                                         4                            7   \n",
      "MS                                            0                            2   \n",
      "ICL                                           3                            4   \n",
      "GCL                                           2                            2   \n",
      "TSR                                           7                           11   \n",
      "\n",
      "            Isochrysis-sp-CCMP1244  Isochrysis-sp-CCMP1324  \\\n",
      "SLC4                             8                       5   \n",
      "Bestrophin                      35                      18   \n",
      "CA_alpha                         0                       0   \n",
      "CA_beta                          5                       4   \n",
      "CA_delta                         1                       2   \n",
      "CA_zeta                          0                       0   \n",
      "PK                               3                       4   \n",
      "PEPC                             1                       0   \n",
      "PEPCK                            1                       1   \n",
      "MDH                              4                       3   \n",
      "OMT                             61                      28   \n",
      "ME                               3                       3   \n",
      "PPDK                             0                       0   \n",
      "PYC                             11                       9   \n",
      "PGP                              7                       4   \n",
      "GOX                              1                       1   \n",
      "SPT                              2                       4   \n",
      "ALAT_GGAT                       13                       8   \n",
      "GDCT                             4                       1   \n",
      "SHMT                             4                       6   \n",
      "HR                               6                       4   \n",
      "GK                               0                       0   \n",
      "GlcDH                            4                       2   \n",
      "MS                               1                       1   \n",
      "ICL                              4                       2   \n",
      "GCL                              4                       1   \n",
      "TSR                              9                       3   \n",
      "\n",
      "            Pleurochrysis-carterae-CCMP645  Pavlova-sp-CCMP459  \\\n",
      "SLC4                                     4                   4   \n",
      "Bestrophin                              14                  19   \n",
      "CA_alpha                                13                   0   \n",
      "CA_beta                                 15                   9   \n",
      "CA_delta                                 0                   0   \n",
      "CA_zeta                                  0                   0   \n",
      "PK                                       9                   2   \n",
      "PEPC                                     1                   0   \n",
      "PEPCK                                    2                   2   \n",
      "MDH                                      2                   6   \n",
      "OMT                                     51                  27   \n",
      "ME                                       2                   2   \n",
      "PPDK                                     3                   1   \n",
      "PYC                                     11                   9   \n",
      "PGP                                      3                   3   \n",
      "GOX                                      1                   1   \n",
      "SPT                                      2                   3   \n",
      "ALAT_GGAT                               10                   6   \n",
      "GDCT                                     1                   1   \n",
      "SHMT                                     5                   4   \n",
      "HR                                       8                   3   \n",
      "GK                                       1                   0   \n",
      "GlcDH                                    2                   5   \n",
      "MS                                       0                   1   \n",
      "ICL                                      2                   2   \n",
      "GCL                                      1                   4   \n",
      "TSR                                      7                   5   \n",
      "\n",
      "            Prymnesium-parvum-Texoma1  Emiliania-huxleyi-374  \\\n",
      "SLC4                               17                      8   \n",
      "Bestrophin                         32                     14   \n",
      "CA_alpha                            0                      0   \n",
      "CA_beta                             5                      0   \n",
      "CA_delta                            6                      1   \n",
      "CA_zeta                             0                      0   \n",
      "PK                                  6                      3   \n",
      "PEPC                                2                      1   \n",
      "PEPCK                               5                      1   \n",
      "MDH                                 4                      3   \n",
      "OMT                                51                     22   \n",
      "ME                                  1                      4   \n",
      "PPDK                                0                      0   \n",
      "PYC                                15                      9   \n",
      "PGP                                 4                      4   \n",
      "GOX                                 2                      1   \n",
      "SPT                                 4                      1   \n",
      "ALAT_GGAT                          14                      7   \n",
      "GDCT                                4                      2   \n",
      "SHMT                                5                      4   \n",
      "HR                                  7                      4   \n",
      "GK                                  0                      0   \n",
      "GlcDH                               5                      4   \n",
      "MS                                  1                      1   \n",
      "ICL                                 6                      3   \n",
      "GCL                                 1                      1   \n",
      "TSR                                 9                      4   \n",
      "\n",
      "            Emiliania-huxleyi-379  Emiliania-huxleyi-CCMP370  \\\n",
      "SLC4                            9                          6   \n",
      "Bestrophin                     22                         38   \n",
      "CA_alpha                        0                          3   \n",
      "CA_beta                         5                          0   \n",
      "CA_delta                        1                          1   \n",
      "CA_zeta                         0                          0   \n",
      "PK                              3                          5   \n",
      "PEPC                            1                          1   \n",
      "PEPCK                           1                          1   \n",
      "MDH                             3                          5   \n",
      "OMT                            35                         73   \n",
      "ME                              2                          2   \n",
      "PPDK                            0                          0   \n",
      "PYC                             9                         13   \n",
      "PGP                             3                          8   \n",
      "GOX                             1                          1   \n",
      "SPT                             0                          3   \n",
      "ALAT_GGAT                      11                         15   \n",
      "GDCT                            3                          4   \n",
      "SHMT                            5                          6   \n",
      "HR                              5                          7   \n",
      "GK                              0                          0   \n",
      "GlcDH                           2                          4   \n",
      "MS                              1                          1   \n",
      "ICL                             2                          8   \n",
      "GCL                             3                          3   \n",
      "TSR                             6                          6   \n",
      "\n",
      "            Emiliania-huxleyi-PLYM219  \n",
      "SLC4                                8  \n",
      "Bestrophin                         32  \n",
      "CA_alpha                            4  \n",
      "CA_beta                             0  \n",
      "CA_delta                            2  \n",
      "CA_zeta                             0  \n",
      "PK                                  5  \n",
      "PEPC                                1  \n",
      "PEPCK                               1  \n",
      "MDH                                 2  \n",
      "OMT                                77  \n",
      "ME                                  2  \n",
      "PPDK                                0  \n",
      "PYC                                16  \n",
      "PGP                                 7  \n",
      "GOX                                 1  \n",
      "SPT                                 2  \n",
      "ALAT_GGAT                          12  \n",
      "GDCT                                3  \n",
      "SHMT                                5  \n",
      "HR                                  6  \n",
      "GK                                  0  \n",
      "GlcDH                               4  \n",
      "MS                                  1  \n",
      "ICL                                 4  \n",
      "GCL                                 4  \n",
      "TSR                                 9  \n"
     ]
    }
   ],
   "source": [
    "a=gen_type('BP')\n",
    "b=gen_type('CF')\n",
    "c=gen_type('PR')\n",
    "\n",
    "framesX=(a,b,c)\n",
    "outFrame=pd.concat(framesX)\n",
    "\n",
    "outFrame.to_csv('COP_ADD.csv')\n",
    "\n",
    "print outFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SLC4           5\n",
      "Bestrophin    48\n",
      "CA_alpha       0\n",
      "CA_beta        2\n",
      "CA_delta       4\n",
      "CA_zeta        0\n",
      "PK             9\n",
      "PEPC           1\n",
      "PEPCK          1\n",
      "MDH            2\n",
      "OMT           47\n",
      "ME             3\n",
      "PPDK           0\n",
      "PYC           10\n",
      "PGP           11\n",
      "GOX            0\n",
      "SPT            3\n",
      "ALAT_GGAT      8\n",
      "GDCT           1\n",
      "SHMT           4\n",
      "HR             9\n",
      "GK             0\n",
      "GlcDH          4\n",
      "MS             0\n",
      "ICL            3\n",
      "GCL            2\n",
      "TSR            7\n",
      "Name: Chrysochromulina-polylepis-CCMP1757, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print outFrame['Chrysochromulina-polylepis-CCMP1757']"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
