{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import sys\n",
    "from ftplib import FTP #import the ftp library\n",
    "import re \n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#t='Dinophyceae'\n",
    "#t='Bacillariophyta'\n",
    "#t='Haptophyta'\n",
    "t='Raphidophyceae'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Chattonella-subsalsa-CCMP2191.pep.fa.gz', 'Heterosigma-akashiwo-CCMP2393.pep.fa.gz', 'Heterosigma-akashiwo-CCMP3107.pep.fa.gz', 'Heterosigma-akashiwo-CCMP452.pep.fa.gz', 'Heterosigma-akashiwo-NB.pep.fa.gz']\n"
     ]
    }
   ],
   "source": [
    "##retrive files\n",
    "mt=open('mmetsp_taxonomy.txt','r')\n",
    "g=[] #make an empty list to store genus names\n",
    "for line in mt:\n",
    "    if re.search(t,line): #if taxa name in line\n",
    "        g= g+line.split('\\t')[7:8]#pull out the 8th field should be genus, keeping as list\n",
    "\n",
    "g=set(g) #keep only unique genus names\n",
    "#print g\n",
    "#close the taxonomy file\n",
    "mt.close()\n",
    "\n",
    "ftp= FTP('ftp.imicrobe.us') #set home ftp server\n",
    "ftp.login() #log in\n",
    "ftp.cwd('camera/combined_assemblies') #ch\n",
    "\n",
    "files=ftp.nlst() #make a list of all files and directories in wd\n",
    "delimiter=' '\n",
    "all=delimiter.join(files)\n",
    "\n",
    "names=[]\n",
    "\n",
    "for genus in g:\n",
    "    string= genus+\"\\S*.pep.fa.gz\"\n",
    "    taxafiles=re.findall(string, all)\n",
    "    #print \"{} files matching genus=\".format(len(taxafiles))+genus\n",
    "    #print taxafiles\n",
    "    if len(taxafiles) > 0:\n",
    "        for filex in taxafiles:\n",
    "            command = \"RETR \"+filex\n",
    "            outfile = filex\n",
    "            #ftp.retrbinary(command, open(outfile, 'wb').write)\n",
    "            names.append(outfile)\n",
    "            \n",
    "ftp.quit()\n",
    "\n",
    "if t=='Dinophyceae':\n",
    "    names.remove('Durinskia-baltica-CSIRO_CS-38.pep.fa.gz')\n",
    "    names.remove('Oxyrrhis-marina-CCMP1795.pep.fa.gz')\n",
    "    names.remove('Alexandrium-fundyense-CCMP1719.pep.fa.gz')\n",
    "\n",
    "print names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Karenia-brevis-CCMP2229.pep.fa.gz', 'Karenia-brevis-SP1.pep.fa.gz', 'Karenia-brevis-SP3.pep.fa.gz']\n"
     ]
    }
   ],
   "source": [
    "#names=names[:3]\n",
    "#print names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def readHMM(Organism,gene_name):\n",
    "    \"\"\" Takes in organism and gene_name from HMM results and makes a table.\n",
    "    HMM results from --tblout that have the following name organism_genename_HMM.csv\n",
    "    Note: pep.fa files differ in structure and it can affect how the HMM output is written. If you can't read the file in \n",
    "    modify the fuction\"\"\"\n",
    "\n",
    "    hold=pd.read_csv('{}_{}_HMM.csv'.format(Organism,gene_name),delim_whitespace=True,engine='python',skiprows=2,skipfooter=10)\n",
    "    #empty files have 12 rows so the following if statement will only work on files that are not empty\n",
    "    \n",
    "    if hold.shape[0]!=0:\n",
    "        readX=pd.read_csv('{}_{}_HMM.csv'.format(Organism,gene_name),delim_whitespace=True,engine='python',skiprows=3,\n",
    "                          header=None ,skipfooter=10)\n",
    "        \n",
    "        samplenames=[]\n",
    "        for i in np.arange(1,readX.shape[1]+1):\n",
    "            samplenames.append(str(i))\n",
    "\n",
    "        readX.columns=samplenames\n",
    "        \n",
    "        new=pd.DataFrame()\n",
    "        new['CAMPEPid']=readX['1']\n",
    "        new['contig']=readX['19']\n",
    "        new['Evalue']=readX['5']\n",
    "        new['Annotation']='{}'.format(gene_name)\n",
    "\n",
    "        new.contig=new.contig.str.split(\"|\").str[1]\n",
    "        return new\n",
    "        #return readX"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "##MAKEHMM\n",
    "def makeHMM(Organism,gene_type):\n",
    "    \"\"\"Combines the plots from each gene_type into a table\"\"\"\n",
    "    if gene_type=='CF':\n",
    "        a=readHMM(Organism,'PK')\n",
    "        b=readHMM(Organism,'PEPC')\n",
    "        c=readHMM(Organism,'PEPCK')\n",
    "        d=readHMM(Organism,'PK')\n",
    "        e=readHMM(Organism,'MDH')\n",
    "        f=readHMM(Organism,'OMT')\n",
    "        g=readHMM(Organism,'PYC')\n",
    "        h=readHMM(Organism,'PPDK')\n",
    "        i=readHMM(Organism,'ME')\n",
    "        \n",
    "        frames = [a,b,c,d,e,f,g,i,h]\n",
    "        result = pd.concat(frames)\n",
    "        return result\n",
    "    if gene_type=='PR':\n",
    "        a=readHMM(Organism,'SHMT')\n",
    "        b=readHMM(Organism,'GOX')\n",
    "        c=readHMM(Organism,'GDCT')\n",
    "        d=readHMM(Organism,'PGP')\n",
    "        e=readHMM(Organism,'ICL')\n",
    "        f=readHMM(Organism,'GCL')\n",
    "        g=readHMM(Organism,'HR')\n",
    "        h=readHMM(Organism,'SPT')\n",
    "        i=readHMM(Organism,'TSR')\n",
    "        j=readHMM(Organism,'MS')\n",
    "        k=readHMM(Organism,'GlcDH')\n",
    "        l=readHMM(Organism,'ALAT_GGAT')\n",
    "        m=readHMM(Organism,'GK')\n",
    "        \n",
    "        frames = [a,b,c,d,e,f,g,h,i,j,k,l,m]\n",
    "        result = pd.concat(frames)\n",
    "        return result\n",
    "    if gene_type=='BP':\n",
    "        a=readHMM(Organism,'CA_alpha')\n",
    "        b=readHMM(Organism,'CA_delta')\n",
    "        c=readHMM(Organism,'CA_beta')\n",
    "        d=readHMM(Organism,'Ca_zeta')\n",
    "        e=readHMM(Organism,'Bestrophin')\n",
    "        f=readHMM(Organism,'SLC4')\n",
    "\n",
    "        frames = [a,b,c,d,e,f]\n",
    "        result = pd.concat(frames)\n",
    "        return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "###COUNTS\n",
    "def HMMcmp(Organism,gene_type):\n",
    "    result1=makeHMM(Organism,gene_type)\n",
    "    contig_EV=dict(zip(result1.contig,result1.Evalue))\n",
    "    contig_Annot=dict(zip(result1.contig,result1.Annotation))\n",
    "    \n",
    "    import re\n",
    "    Organism = re.sub('.pep.fa.gz', '', Organism)\n",
    "    \n",
    "    expression=pd.read_csv('{}_cds_counts.txt'.format(Organism),delimiter='\\t',index_col=0)\n",
    "    \n",
    "    #treatments=len(list(expression))\n",
    "    #print treatments\n",
    "    \n",
    "    expression['log2CPM']=np.log2(expression.sum(axis=1)/1000000)\n",
    "    expression.index= expression.index.str.split(\"|\").str[1]\n",
    "    contig_CMP=dict(zip(expression.index,expression.log2CPM))\n",
    "    \n",
    "    Contig=[]\n",
    "    Evalue=[]\n",
    "    Annotation=[]\n",
    "    log2CPM=[]\n",
    "    \n",
    "    for i in expression.index:\n",
    "        a=contig_CMP.get(i)\n",
    "        if a>-16 or a==\"-inf\" or a==\"inf\":\n",
    "            Contig.append(i)\n",
    "            Evalue.append(contig_EV.get(i))\n",
    "            Annotation.append(contig_Annot.get(i))\n",
    "            log2CPM.append(a)\n",
    "    \n",
    "    out=pd.DataFrame()\n",
    "    out['Contig']=Contig\n",
    "    out['Evalue']=Evalue\n",
    "    out['Annotation']=Annotation\n",
    "    out['log2CPM']=log2CPM\n",
    "        \n",
    "    return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Contig</th>\n",
       "      <th>Evalue</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>log2CPM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>267033_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-12.137153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3814_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-9.180025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>77900_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-11.266233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>163778_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-13.609640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>156264_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-15.231129</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Contig  Evalue Annotation    log2CPM\n",
       "0  267033_1     NaN       None -12.137153\n",
       "1    3814_1     NaN       None  -9.180025\n",
       "2   77900_1     NaN       None -11.266233\n",
       "3  163778_1     NaN       None -13.609640\n",
       "4  156264_1     NaN       None -15.231129"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HMMcmp('Heterosigma-akashiwo-CCMP2393.pep.fa.gz','BP').head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "###CLEAN\n",
    "def HMMclean(Organism):\n",
    "    '''Will create an HMM table and filter out values that do not meet an e-value'''\n",
    "    aX=HMMcmp(Organism,'BP')\n",
    "    bX=HMMcmp(Organism,'CF')\n",
    "    cX=HMMcmp(Organism,'PR')\n",
    "    \n",
    "    frames=[aX,bX,cX]\n",
    "    HMM= pd.concat(frames)\n",
    "     \n",
    "    HMM=HMM[HMM['Evalue'] < .0001]\n",
    "    \n",
    "    #return HMM.drop_duplicates()\n",
    "    \n",
    "    Genes= np.unique(HMM['Annotation'])\n",
    "    AX=pd.DataFrame(0,index=[Organism],columns=Genes)\n",
    "    \n",
    "    for j in Genes:\n",
    "        counts=HMM.Annotation.value_counts()[j]\n",
    "        AX[j]= counts\n",
    "           \n",
    "    return AX\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ALAT_GGAT</th>\n",
       "      <th>Bestrophin</th>\n",
       "      <th>CA_alpha</th>\n",
       "      <th>CA_beta</th>\n",
       "      <th>GCL</th>\n",
       "      <th>GDCT</th>\n",
       "      <th>GK</th>\n",
       "      <th>GOX</th>\n",
       "      <th>GlcDH</th>\n",
       "      <th>HR</th>\n",
       "      <th>...</th>\n",
       "      <th>ME</th>\n",
       "      <th>OMT</th>\n",
       "      <th>PEPCK</th>\n",
       "      <th>PGP</th>\n",
       "      <th>PK</th>\n",
       "      <th>PYC</th>\n",
       "      <th>SHMT</th>\n",
       "      <th>SLC4</th>\n",
       "      <th>SPT</th>\n",
       "      <th>TSR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Heterosigma-akashiwo-CCMP2393.pep.fa.gz</th>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>62</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         ALAT_GGAT  Bestrophin  CA_alpha  \\\n",
       "Heterosigma-akashiwo-CCMP2393.pep.fa.gz         13           6         5   \n",
       "\n",
       "                                         CA_beta  GCL  GDCT  GK  GOX  GlcDH  \\\n",
       "Heterosigma-akashiwo-CCMP2393.pep.fa.gz        9    1     1   1    4      2   \n",
       "\n",
       "                                         HR ...   ME  OMT  PEPCK  PGP  PK  \\\n",
       "Heterosigma-akashiwo-CCMP2393.pep.fa.gz   5 ...    4   62     13    9   3   \n",
       "\n",
       "                                         PYC  SHMT  SLC4  SPT  TSR  \n",
       "Heterosigma-akashiwo-CCMP2393.pep.fa.gz   15     4     3    1    6  \n",
       "\n",
       "[1 rows x 22 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HMMclean('Heterosigma-akashiwo-CCMP2393.pep.fa.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "###GENE_TYPE\n",
    "def gen_type(gene_type):\n",
    "    \"\"\" Allows you to extract specific gene_types\"\"\"\n",
    "    \n",
    "    if gene_type=='BP':\n",
    "            fixed=['SLC4','Bestrophin','CA_alpha','CA_beta','CA_delta','CA_zeta']\n",
    "    if gene_type=='PR':\n",
    "            fixed=['PGP','GOX','SPT','ALAT_GGAT','GDCT','SHMT','HR','GK','GlcDH','MS','ICL','GCL','TSR']\n",
    "    if gene_type=='CF':\n",
    "            fixed=['PK','PEPC','PEPCK','MDH','OMT','ME','PPDK','PYC']\n",
    "            \n",
    "    A=pd.DataFrame(0, index=names, columns=fixed)\n",
    "    \n",
    "    for i,j in enumerate(names):\n",
    "        for k in fixed:\n",
    "            B=HMMclean(j)\n",
    "            #print B\n",
    "            if k in list(B):\n",
    "                A[k][i]=B[k]\n",
    "                \n",
    "    A.index = A.index.str.split('.').str[0] \n",
    "    ####the count is the number of transcriptomes taken into account\n",
    "    ###I just added gene_name to teh count to make teh values unique for when teh different gene_type frames are concatenated\n",
    "    A['count{}'.format(gene_type)]=0\n",
    "    \n",
    "    for i,j in enumerate(A.index):\n",
    "        expression=pd.read_csv('{}_cds_counts.txt'.format(j),delimiter='\\t',index_col=0)\n",
    "        A['count{}'.format(gene_type)][i]=len(list(expression))\n",
    "                \n",
    "    C=A.transpose()\n",
    "    C.to_csv('{}_{}_GeneCountHMM.csv'.format(t,gene_type))\n",
    "    \n",
    "    return A.transpose()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Karenia-brevis-CCMP2229</th>\n",
       "      <th>Karenia-brevis-SP1</th>\n",
       "      <th>Karenia-brevis-SP3</th>\n",
       "      <th>Karenia-brevis-Wilson</th>\n",
       "      <th>Peridinium-aciculiferum-PAER_2</th>\n",
       "      <th>Karlodinium-micrum-CCMP2283</th>\n",
       "      <th>Prorocentrum-minimum-CCMP1329</th>\n",
       "      <th>Prorocentrum-minimum-CCMP2233</th>\n",
       "      <th>Symbiodinium-kawagutii-CCMP2468</th>\n",
       "      <th>Symbiodinium-sp-C1</th>\n",
       "      <th>...</th>\n",
       "      <th>Alexandrium-monilatum-CCMP3105</th>\n",
       "      <th>Alexandrium-temarense-CCMP1771</th>\n",
       "      <th>Azadinium-spinosum-3D9</th>\n",
       "      <th>Ceratium-fusus-PA161109</th>\n",
       "      <th>Oxyrrhis-marina-LB1974</th>\n",
       "      <th>Oxyrrhis-marina</th>\n",
       "      <th>Glenodinium-foliaceum-CCAP1116_3</th>\n",
       "      <th>Scrippsiella-hangoei-SHTV5</th>\n",
       "      <th>Scrippsiella-hangoei_like-SHHI_4</th>\n",
       "      <th>Scrippsiella-trochoidea-CCMP3099</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SLC4</th>\n",
       "      <td>12</td>\n",
       "      <td>19</td>\n",
       "      <td>14</td>\n",
       "      <td>32</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bestrophin</th>\n",
       "      <td>132</td>\n",
       "      <td>139</td>\n",
       "      <td>90</td>\n",
       "      <td>128</td>\n",
       "      <td>27</td>\n",
       "      <td>78</td>\n",
       "      <td>49</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>...</td>\n",
       "      <td>86</td>\n",
       "      <td>85</td>\n",
       "      <td>62</td>\n",
       "      <td>87</td>\n",
       "      <td>54</td>\n",
       "      <td>59</td>\n",
       "      <td>71</td>\n",
       "      <td>61</td>\n",
       "      <td>57</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CA_alpha</th>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>25</td>\n",
       "      <td>20</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>23</td>\n",
       "      <td>19</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>35</td>\n",
       "      <td>29</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CA_beta</th>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>15</td>\n",
       "      <td>18</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CA_delta</th>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CA_zeta</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>countBP</th>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Karenia-brevis-CCMP2229  Karenia-brevis-SP1  Karenia-brevis-SP3  \\\n",
       "SLC4                             12                  19                  14   \n",
       "Bestrophin                      132                 139                  90   \n",
       "CA_alpha                         20                  19                  19   \n",
       "CA_beta                           7                   6                   5   \n",
       "CA_delta                          9                   7                   8   \n",
       "CA_zeta                           0                   0                   0   \n",
       "countBP                           4                   2                   2   \n",
       "\n",
       "            Karenia-brevis-Wilson  Peridinium-aciculiferum-PAER_2  \\\n",
       "SLC4                           32                               2   \n",
       "Bestrophin                    128                              27   \n",
       "CA_alpha                       19                              25   \n",
       "CA_beta                         4                              18   \n",
       "CA_delta                        7                               0   \n",
       "CA_zeta                         0                               0   \n",
       "countBP                         4                               2   \n",
       "\n",
       "            Karlodinium-micrum-CCMP2283  Prorocentrum-minimum-CCMP1329  \\\n",
       "SLC4                                  3                              2   \n",
       "Bestrophin                           78                             49   \n",
       "CA_alpha                             20                              7   \n",
       "CA_beta                               0                              0   \n",
       "CA_delta                              6                              5   \n",
       "CA_zeta                               0                              0   \n",
       "countBP                               3                              4   \n",
       "\n",
       "            Prorocentrum-minimum-CCMP2233  Symbiodinium-kawagutii-CCMP2468  \\\n",
       "SLC4                                    2                                1   \n",
       "Bestrophin                             50                                0   \n",
       "CA_alpha                                6                                0   \n",
       "CA_beta                                 0                                0   \n",
       "CA_delta                                7                                0   \n",
       "CA_zeta                                 0                                0   \n",
       "countBP                                 3                                4   \n",
       "\n",
       "            Symbiodinium-sp-C1                ...                 \\\n",
       "SLC4                         2                ...                  \n",
       "Bestrophin                  32                ...                  \n",
       "CA_alpha                     7                ...                  \n",
       "CA_beta                     22                ...                  \n",
       "CA_delta                     1                ...                  \n",
       "CA_zeta                      0                ...                  \n",
       "countBP                      2                ...                  \n",
       "\n",
       "            Alexandrium-monilatum-CCMP3105  Alexandrium-temarense-CCMP1771  \\\n",
       "SLC4                                     4                              10   \n",
       "Bestrophin                              86                              85   \n",
       "CA_alpha                                19                              23   \n",
       "CA_beta                                  8                               8   \n",
       "CA_delta                                 6                               7   \n",
       "CA_zeta                                  0                               0   \n",
       "countBP                                  4                               4   \n",
       "\n",
       "            Azadinium-spinosum-3D9  Ceratium-fusus-PA161109  \\\n",
       "SLC4                             6                        5   \n",
       "Bestrophin                      62                       87   \n",
       "CA_alpha                        19                       24   \n",
       "CA_beta                          7                       20   \n",
       "CA_delta                         2                        0   \n",
       "CA_zeta                          0                        0   \n",
       "countBP                          3                        2   \n",
       "\n",
       "            Oxyrrhis-marina-LB1974  Oxyrrhis-marina  \\\n",
       "SLC4                             0                1   \n",
       "Bestrophin                      54               59   \n",
       "CA_alpha                        16               17   \n",
       "CA_beta                          9               15   \n",
       "CA_delta                         0                0   \n",
       "CA_zeta                          0                0   \n",
       "countBP                          3                4   \n",
       "\n",
       "            Glenodinium-foliaceum-CCAP1116_3  Scrippsiella-hangoei-SHTV5  \\\n",
       "SLC4                                       2                           3   \n",
       "Bestrophin                                71                          61   \n",
       "CA_alpha                                  35                          29   \n",
       "CA_beta                                   18                          23   \n",
       "CA_delta                                   1                           0   \n",
       "CA_zeta                                    0                           0   \n",
       "countBP                                    2                           3   \n",
       "\n",
       "            Scrippsiella-hangoei_like-SHHI_4  Scrippsiella-trochoidea-CCMP3099  \n",
       "SLC4                                       2                                 0  \n",
       "Bestrophin                                57                                 2  \n",
       "CA_alpha                                  30                                 0  \n",
       "CA_beta                                   28                                 0  \n",
       "CA_delta                                   0                                 0  \n",
       "CA_zeta                                    0                                 0  \n",
       "countBP                                    3                                 3  \n",
       "\n",
       "[7 rows x 27 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gen_type('BP')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COMPLETE\n"
     ]
    }
   ],
   "source": [
    "a=gen_type('BP')\n",
    "b=gen_type('CF')\n",
    "c=gen_type('PR')\n",
    "\n",
    "framesX=(a,b,c)\n",
    "outFrame=pd.concat(framesX)\n",
    "\n",
    "outFrame.to_csv('{}_GeneCountHMM.csv'.format(t))\n",
    "\n",
    "print 'COMPLETE'"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
